src/interactions/panZoomInteraction.ts

```diff
@@ -20,56 +29,143 @@ export module Interaction {
      */
     constructor(xScale?: Scale.AbstractQuantitative<any>, yScale?: Scale.AbstractQuantitative<any>) {
       super();
-      if (xScale) {
-        this._xScale = xScale;
-        // HACKHACK #1388: self-register for resetZoom()
-        this._xScale.broadcaster.registerListener("pziX" + this.getID(), () => this.resetZoom());
-      }
-      if (yScale) {
-        this._yScale = yScale;
-        // HACKHACK #1388: self-register for resetZoom()
-        this._yScale.broadcaster.registerListener("pziY" + this.getID(), () => this.resetZoom());
+      this._xScale = xScale;
+      this._yScale = yScale;
+
+      this._dragInteraction = new Interaction.Drag();
+      this._setupDragInteraction();
+      this._touchIds = d3.map();
+    }
+
+    public _anchor(component: Component.AbstractComponent, hitBox: D3.Selection) {
+      super._anchor(component, hitBox);
+      this._dragInteraction._anchor(component, hitBox);
+
+      var mouseDispatcher = Dispatcher.Mouse.getDispatcher(<SVGElement> this._componentToListenTo.content().node());
+      mouseDispatcher.onWheel("Interaction.PanZoom" + this.getID(), (p: Point, e: WheelEvent) => this._handleWheelEvent(p, e));
+
+      this._touchDispatcher = Dispatcher.Touch.getDispatcher(<SVGElement> this._componentToListenTo.content().node());
+      this._touchDispatcher.onTouchStart("Interaction.PanZoom" + this.getID(),
+        (ids, idToPoint, e) => this._handleTouchStart(ids, idToPoint, e));
+      this._touchDispatcher.onTouchMove("Interaction.PanZoom" + this.getID(),
+        (ids, idToPoint, e) => this._handleTouchMove(ids, idToPoint, e));
+      this._touchDispatcher.onTouchEnd("Interaction.PanZoom" + this.getID(),
+        (ids, idToPoint, e) => this._handleTouchEnd(ids, idToPoint, e));
+    }
+
+    private _handleTouchStart(ids: number[], idToPoint: { [id: number]: Point; }, e: TouchEvent) {
+      ids.forEach((id) => {
+        if (this._touchIds.size() === 2) {
+          return;
+        }
+
+        this._touchIds.set(id.toString(), this._translateToComponentSpace(idToPoint[id]));
+      });
+      if (this._touchIds.size() === 2) {
+        this._dragInteraction.onDrag(null);
```

Comment:
Instead of blanking the `onDrag` callback, would it make more sense to have some sort of conditional-check inside the callback that skips actions if there is more than one touch?

Might also be good to label this with a `HACKHACK` on #696.